cpp/notebooks/ros1_widget_tutorial.ipynb

```diff
@@ -0,0 +1,310 @@
+   "outputs": [],
+   "source": [
+    "#include \"jupyter_ros_utils/ros1_publisher.hpp\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5c016d7f-6f88-4332-94f9-31c1bf9d4dc7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ros_init(\"cpp_ros_async_callback_publisher\");"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "43ea2cb3-3621-4b0a-ba99-6e8969c35a76",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#include <std_msgs/UInt32.h>\n",
+    "\n",
+    "uint32_t data = 0;"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1e43516a-ba56-4273-8535-13e7089ffe2b",
+   "metadata": {},
+   "source": [
+    "### **Initialize ROS Spinner and start publisher**\n",
+    "This widget will use publish data using pub_callback and show it in the stream"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "db89e15f-55f8-4ba9-b8d6-82d3ef53c539",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ros_init(\"cpp_ros_async_callback_publisher\");"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "de86e5ae-6589-4491-87ef-e93f40b7a001",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "std_msgs::UInt32 pub_callback() {\n",
+    "    std_msgs::UInt32 msg;\n",
+    "    msg.data = data;\n",
+    "    ++data;\n",
+    "    ROS_INFO(\"%d\", data);\n",
+    "    \n",
+    "    return msg;\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "705fc97c-a411-4ad7-b0ba-f42c23d2b1ec",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ZeusPublisher pub(\"/magic/uint\", pub_callback);"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "29d73b56-f54f-4c64-9577-2bb4ec499cb4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! ros::shutdown()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d339170f-c43a-42eb-988d-df9457fdc42c",
+   "metadata": {},
+   "source": [
+    "## **ROS Subscriber**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f2d92037-0d68-4417-bf1e-f11becf9d60c",
+   "metadata": {},
+   "source": [
+    "### **Load ROS Subscriber**."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "de5f1335-9084-44dd-9755-c4d96712aac8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#include \"jupyter_ros_utils/ros1_subscriber.hpp\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0c990c19-e277-4ce8-835e-4596da115194",
+   "metadata": {},
+   "source": [
+    "### **Initialize ROS Spinner and start subscriber.**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "05a244c4-c122-4e58-9979-3b6298976ff7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#include <std_msgs/String.h>\n",
+    "\n",
+    "void cb(const std_msgs::String& msg) {\n",
+    "    std::cout << msg << \"\\n\";\n",
+    "}\n",
```

Comment:
As far as I remember, function declaration in xeus-cling need to be in its own cell. Have you tried running this in the notebook?